Server EXTERNAL_WMS parameter not working

### What is the bug or the crash?

When using EXTERNAL_WMS in Servers GetPrint or GetMap requests, external layer is not working.  I checked current master and 3.22.5.

**MASTER**
Result is XML Exception LayerNotDefined

Unit tests `PyQgsServerWMSGetMap (test_wms_getmap_external)` and `PyQgsServerWMSGetPrint (test_wms_getprint_external)` are also failing with same XML error.

```
Traceback (most recent call last):
  File "/home/uros/dev/cpp/QGIS/tests/src/python/test_qgsserver_wms_getmap.py", line 1694, in test_wms_getmap_external
    self._img_diff_error(r, h, "WMS_GetMap_External", 20000)
  File "/home/uros/dev/cpp/QGIS/tests/src/python/test_qgsserver.py", line 235, in _img_diff_error
    self.assertEqual(
AssertionError: 'text/xml; charset=utf-8' != 'image/png'
- text/xml; charset=utf-8
+ image/png
 : Content type is wrong: text/xml; charset=utf-8 instead of image/png
b'<ServiceExceptionReport xmlns="http://www.opengis.net/ogc" version="1.3.0">\n <ServiceException code="LayerNotDefined">The layer \'landsat\' does not exist.</ServiceException>\n</ServiceExceptionReport>
```

**3.22.5**
You get the map but without external layer, no errors

### Steps to reproduce the issue

Add this project and data to /tests/testdata/qgis_server_external/
[ext_wms_issue.zip](https://github.com/qgis/QGIS/files/8725339/ext_wms_issue.zip)

Test GetMap and GetPrint requests (adjust your path)
[GetMap](http://localhost:8000/?map=/home/uros/dev/cpp/QGIS/tests/testdata/qgis_server_external/ext_wms_issue.qgs&LAYERS=EXTERNAL_WMS%3AOpenLayers_Layer_WMS_20%2Cpolys_with_id_b2dbe530_e0ab_46d2_b070_b1220011f016&FORMAT=image%2Fpng&TRANSPARENT=TRUE&DPI=96&VERSION=1.3.0&EXCEPTIONS=XML&SERVICE=WMS&REQUEST=GetMap&STYLES=&CRS=EPSG%3A3857&BBOX=-12800626.67562,3248186.0973439,-7698302.164238,5116918.5645997&WIDTH=1043&HEIGHT=382&OpenLayers_Layer_WMS_20%3Aurl=https%3A%2F%2Fwww.gebco.net%2Fdata_and_products%2Fgebco_web_services%2Fweb_map_service%2Fmapserv%3F&OpenLayers_Layer_WMS_20%3Aformat=image%2Fpng&OpenLayers_Layer_WMS_20%3Acrs=EPSG%3A3857&OpenLayers_Layer_WMS_20%3Alayers=GEBCO_LATEST_2&OpenLayers_Layer_WMS_20%3Astyles=
)
[GetPrint](http://localhost:8000/?map=/home/uros/dev/cpp/QGIS/tests/testdata/qgis_server_external/ext_wms_issue.qgs&SERVICE=WMS&VERSION=1.3&REQUEST=GetPrint&FORMAT=png&EXCEPTIONS=application%2Fvnd.ogc.se_inimage&TRANSPARENT=true&SRS=EPSG%3A3857&DPI=150&TEMPLATE=a4%20test&map0%3Aextent=-11624780.5%2C3430129.8%2C-9074719.1%2C4700160.3&map0%3Ascale=10000000&map0%3Arotation=0&LAYERS=EXTERNAL_WMS%3AOpenLayers_Layer_WMS_20%2Cpolys_with_id_b2dbe530_e0ab_46d2_b070_b1220011f016&STYLES=&description=&title=&OpenLayers_Layer_WMS_20%3Aurl=https%3A%2F%2Fwww.gebco.net%2Fdata_and_products%2Fgebco_web_services%2Fweb_map_service%2Fmapserv%3F&OpenLayers_Layer_WMS_20%3Aformat=image%2Fpng&OpenLayers_Layer_WMS_20%3Acrs=EPSG%3A3857&OpenLayers_Layer_WMS_20%3Alayers=GEBCO_LATEST_2&OpenLayers_Layer_WMS_20%3Astyles)


### Versions

Master, 3.22.5

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

--- a/src/server/services/wms/qgswmsparameters.cpp
+++ b/src/server/services/wms/qgswmsparameters.cpp
@@ -1971,8 +1971,21 @@ namespace QgsWms
     return param;
   }
 
-  QString QgsWmsParameters::externalWMSUri( const QString &id ) const
+  QString QgsWmsParameters::externalWMSUri( const QString &layerId ) const
   {
+
+    // Param names may be uppercased.
+    QString id { layerId };
+
+    for ( auto it = mExternalWMSParameters.cbegin(); it != mExternalWMSParameters.cend(); ++it )
+    {
+      if ( it.key().compare( id, Qt::CaseSensitivity::CaseInsensitive ) == 0 )
+      {
+        id = it.key();
+        break;
+      }
+    }
+
     if ( !mExternalWMSParameters.contains( id ) )
     {
       return QString();

--- a/src/server/services/wms/qgswmsrendercontext.cpp
+++ b/src/server/services/wms/qgswmsrendercontext.cpp
@@ -819,7 +819,7 @@ bool QgsWmsRenderContext::isExternalLayer( const QString &name ) const
 {
   for ( const auto &layer : mExternalLayers )
   {
-    if ( layer->name().compare( name ) == 0 )
+    if ( layerNickname( *layer ).compare( name ) == 0 )
       return true;
   }
 

--- a/src/server/services/wms/qgswmsrendercontext.cpp
+++ b/src/server/services/wms/qgswmsrendercontext.cpp
@@ -518,6 +518,7 @@ void QgsWmsRenderContext::searchLayersToRenderStyle()
 
     if ( ! param.mExternalUri.isEmpty() && ( mFlags & AddExternalLayers ) )
     {
+
       std::unique_ptr<QgsMapLayer> layer = std::make_unique< QgsRasterLayer >( param.mExternalUri, param.mNickname, QStringLiteral( "wms" ) );
 
       if ( layer->isValid() )

--- a/tests/src/server/wms/test_qgsserver_wms_parameters.cpp
+++ b/tests/src/server/wms/test_qgsserver_wms_parameters.cpp
@@ -76,6 +76,26 @@ void TestQgsServerWmsParameters::external_layers()
   QCOMPARE( layers_params[0].mOpacity, 255 );
   QCOMPARE( layers_params[1].mOpacity, 200 );
   QCOMPARE( layers_params[2].mOpacity, 125 );
+
+  // Upper case
+  QUrlQuery query2;
+  query2.addQueryItem( "LAYERS", "EXTERNAL_WMS:external_layer_1,layer,EXTERNAL_WMS:external_layer_2" );
+  query2.addQueryItem( "EXTERNAL_LAYER_1:url", "http://url_1" );
+  query2.addQueryItem( "EXTERNAL_LAYER_1:layers", "layer_1_name" );
+  query2.addQueryItem( "external_layer_2:url", "http://url_2" );
+  query2.addQueryItem( "external_layer_2:layers", "layer_2_name" );
+  query2.addQueryItem( "external_layer_2:opacities", "100" );
+  query2.addQueryItem( "OPACITIES", "255,200,125" );
+
+  const QgsWms::QgsWmsParameters parameters2( query );
+  QList<QgsWms::QgsWmsParametersLayer> layers_params2 = parameters2.layersParameters();
+  QCOMPARE( layers_params2.size(), 3 );
+
+  QgsWms::QgsWmsParametersLayer layer_params2 = layers_params2[0];
+  QCOMPARE( layer_params2.mNickname, QString( "external_layer_1" ) );
+  QCOMPARE( layer_params2.mExternalUri, QString( "layers=layer_1_name&url=http://url_1" ) );
+
+
 }
 
 void TestQgsServerWmsParameters::percent_encoding()

